### PR TITLE
Add documentation about resource profiles

### PR DESCRIPTION
**Pull request by mattwiller**
Adding some documentation about how to use FHIR profiles to control resource validation.  This doesn't yet include anything about authoring profiles: it's complex and we should probably provide tooling for users so they can avoid hand-writing the JSON

### CODE REVIEW

File: packages/docs/docs/fhir-datastore/profiles.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+Using a pre-existing profile is simple: placing the canonical URL of the profile in a resource's `meta.profile` field
+will cause the server to attempt to validate the resource against that profile when the resource is written.
+
+```json
+{
+  "resourceType": "Patient",
+  "meta": {
+    "profile": ["http://hl7.org/fhir/us/core/StructureDefinition/us-core-patient"]
+  },
+  // Normally Patient has no required fields, but under the US Core Patient profile,
```
Comment (ThatOneBro) [inline packages/docs/docs/fhir-datastore/profiles.md:43]: Technically comments in JSON make it invalid, which is why it's highlighted in red on Github. Maybe `jsonc` is a valid formatting here though?

**Review by ThatOneBro — APPROVED** [review]
Looks good to me, I learned a lot about profiles just from reading this!

File: packages/docs/docs/fhir-datastore/profiles.md (hunk 1 above)
Comment (codyebberson) [reply packages/docs/docs/fhir-datastore/profiles.md:43]: Elsewhere we've used just `js` which should do the trick.

File: packages/docs/docs/fhir-datastore/profiles.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+[profiling]: http://hl7.org/fhir/profiling.html
+[us-core-bp]: http://hl7.org/fhir/us/core/StructureDefinition/us-core-blood-pressure
+[us-core]: https://www.hl7.org/fhir/us/core/#us-core-profiles
+[uscdi]: https://www.healthit.gov/isa/united-states-core-data-interoperability-uscdi
+
+## Profile adoption
+
+Using a pre-existing profile is simple: placing the canonical URL of the profile in a resource's `meta.profile` field
+will cause the server to attempt to validate the resource against that profile when the resource is written.
```
Comment (rahul1) [inline packages/docs/docs/fhir-datastore/profiles.md:35]: Great example - I just want to break it down a little for a novice audience. 

- Can you describe which fields are required by the US Core patient resource?
- Can you add an example of the error message that users can expect when creating a `Patient` that _doesn't_ conform to the the profile (having the error message indexed by Algolia will help our users when troubleshooting their apps)
- Lastly, have this example below, to show what the _correct_  data would be

**Review by rahul1 — CHANGES_REQUESTED** [review]

**Review by rahul1 — APPROVED** [review]

**Pull request merged**